common/client_config.go

```diff
@@ -0,0 +1,40 @@
+package common
+
+import (
+	"fmt"
+	"time"
+
+	clients_v2 "github.com/Layr-Labs/eigenda/api/clients/v2"
+)
+
+// ClientConfigV2 contains all non-sensitive configuration to construct V2 clients
+type ClientConfigV2 struct {
+	// Enabled is true if using eigenDA v2, or false if using eigenDA v1
+	Enabled                  bool
+	DisperserClientCfg       clients_v2.DisperserClientConfig
+	PayloadDisperserCfg      clients_v2.PayloadDisperserConfig
+	RelayPayloadRetrieverCfg clients_v2.RelayPayloadRetrieverConfig
+	// hex address of the EigenDAServiceManager contract
+	ServiceManagerAddress string
+	PutRetries            uint
+	// interval at which the ethereum node is polled, while waiting to reach a given block number
+	BlockNumberPollIntervalDuration time.Duration
+	EigenDACertVerifierAddress      string
+}
+
+// Check checks config invariants, and returns an error if there is a problem with the config struct
+func (cfg *ClientConfigV2) Check() error {
+	if cfg.ServiceManagerAddress == "" {
+		return fmt.Errorf("service manager address is required for using EigenDA V2 backend")
+	}
+
+	if cfg.DisperserClientCfg.Hostname == "" {
```

Comment:
Should we also check port too? IIRC it's also a str

Reply:
[f1e4a8c](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/f1e4a8ce604178ff6a0dc8e2962ff3c2cd8cd6c1)